src/hotspot/cpu/aarch64/aarch64_neon_ad.m4

```diff
@@ -147,39 +150,87 @@ instruct vcvt$1$2to$1$3`'(vec$4 dst, vec$5 src)
   %}
   ins_encode %{
-    __ $5(as_FloatRegister($dst$$reg), __ T$7, as_FloatRegister($src$$reg), __ T$6);
-    __ $5(as_FloatRegister($dst$$reg), __ T$9, as_FloatRegister($dst$$reg), __ T$8);
+    __ $6(as_FloatRegister($dst$$reg), __ T$8, as_FloatRegister($src$$reg), __ T$7);
+    __ $6(as_FloatRegister($dst$$reg), __ T$10, as_FloatRegister($dst$$reg), __ T$9);
+  %}
+  ins_pipe(pipe_class_default);
+%}')dnl
+dnl               $1 $2 $3 $4 $5 $6    $7  $8  $9  $10
+VECTOR_CAST_I2I_L(4, I, B, D, X, xtn,  4S, 4H, 8H, 8B)
+VECTOR_CAST_I2I_L(4, B, I, X, D, sxtl, 8B, 8H, 4H, 4S)
+dnl
+instruct vcvt2Lto2F(vecD dst, vecX src)
+%{
+  predicate(n->as_Vector()->length() == 2 && n->bottom_type()->is_vect()->element_basic_type() == T_FLOAT);
+  match(Set dst (VectorCastL2X src));
+  format %{ "scvtfv  T2D, $dst, $src\n\t"
+            "fcvtn   $dst, T2S, $dst, T2D\t# convert 2L to 2F vector"
+  %}
+  ins_encode %{
+    __ scvtfv(__ T2D, as_FloatRegister($dst$$reg), as_FloatRegister($src$$reg));
+    __ fcvtn(as_FloatRegister($dst$$reg), __ T2S, as_FloatRegister($dst$$reg), __ T2D);
   %}
   ins_pipe(pipe_slow);
+%}
+dnl
+define(`VECTOR_CAST_I2F', `
+instruct vcvt$1$2to$1$3`'(vec$4 dst, vec$4 src)
+%{
+  predicate(n->as_Vector()->length() == $1 && n->bottom_type()->is_vect()->element_basic_type() == T_`'TYPE2DATATYPE($3));
+  match(Set dst (VectorCast$2`'2X src));
+  format %{ "scvtfv  T$5, $dst, $src\t# convert $1$2 to $1$3 vector" %}
+  ins_encode %{
+    __ scvtfv(__ T$5, as_FloatRegister($dst$$reg), as_FloatRegister($src$$reg));
+  %}
+  ins_pipe(pipe_class_default);
 %}')dnl
-dnl             $1 $2 $3 $4 $5    $6  $7  $8  $9
-VECTOR_CAST_B2I(B, I, X, D, sxtl, 8B, 8H, 4H, 4S)
-VECTOR_CAST_B2I(I, B, D, X, xtn,  4S, 4H, 8H, 8B)
-
+dnl             $1 $2 $3 $4 $5
+VECTOR_CAST_I2F(2, I, F, D, 2S)
+VECTOR_CAST_I2F(4, I, F, X, 4S)
+VECTOR_CAST_I2F(2, L, D, X, 2D)
+dnl
+define(`VECTOR_CAST_I2F_L', `
+instruct vcvt$1$2to$1$3`'(vec$4 dst, vec$5 src)
+%{
+  predicate(n->as_Vector()->length() == $1 && n->bottom_type()->is_vect()->element_basic_type() == T_`'TYPE2DATATYPE($3));
+  match(Set dst (VectorCast$2`'2X src));
+  format %{ "sxtl    $dst, T$7, $src, T$6\n\t"
+            "scvtfv  T$7, $dst, $dst\t# convert $1$2 to $1$3 vector"
+  %}
+  ins_encode %{
+    __ sxtl(as_FloatRegister($dst$$reg), __ T$7, as_FloatRegister($src$$reg), __ T$6);
+    __ scvtfv(__ T$7, as_FloatRegister($dst$$reg), as_FloatRegister($dst$$reg));
+  %}
+  ins_pipe(pipe_slow);
+%}')dnl
+dnl               $1 $2 $3 $4 $5 $6  $7
+VECTOR_CAST_I2F_L(4, S, F, X, D, 4H, 4S)
+VECTOR_CAST_I2F_L(2, I, D, X, D, 2S, 2D)
+dnl
 instruct vcvt4Bto4F(vecX dst, vecD src)
 %{
   predicate(n->as_Vector()->length() == 4 && n->bottom_type()->is_vect()->element_basic_type() == T_FLOAT);
   match(Set dst (VectorCastB2X src));
-  format %{ "sxtl  $dst, T8H, $src, T8B\n\t"
-            "sxtl  $dst, T4S, $dst, T4H\n\t"
+  format %{ "sxtl    $dst, T8H, $src, T8B\n\t"
+            "sxtl    $dst, T4S, $dst, T4H\n\t"
```

Comment:
This whitespace change looks odd.

Reply:
This whitespace is to align with the next instruction.